client_test.go

```diff
@@ -132,9 +132,12 @@ func TestMain(m *testing.M) {
 				fmt.Fprintln(os.Stderr, "failed to wait for containerd", err)
 			}
 		}
-		if err := os.RemoveAll(defaultRoot); err != nil {
-			fmt.Fprintln(os.Stderr, "failed to remove test root dir", err)
-			os.Exit(1)
+		// this directory is a volume in a container so don't remove
+		if len(os.Getenv("container")) == 0 {
+			if err := os.RemoveAll(defaultRoot); err != nil {
```

Comment:
Maybe just ignore the remove error instead of skipping remove?

If someone uses a bind mount for this directory, or runs a shell and does multiple test runs in the same container,  they'll still want the files cleaned up.